### DISCUSSION TITLE
BI Tool Integration

### DISCUSSION THREAD

**Discussion by paulsp94** [plausible/analytics#2027] [Anything Else]
Is it possible to connect the data from plausible to a BI Tool like Metabase to aggregate the datapoints and create custom KPI's based on the plausible tracking data?

**Comment by metmarkosaric** [comment]
thanks for the feedback @paulsp94! 

we haven't built any direct integrations from our side. we do allows you to take the stats using [CSV export](https://plausible.io/docs/export-stats) and the [stats API](https://plausible.io/docs/stats-api) so perhaps there's a way to import the stats to the tool of your choice with those?

i do know that there is a Plausible connector for Google Data Studio. it is not made by us and it is a paid product, but for those that require it, it could be worth checking out https://www.epilocal.com/products/plausible-data-studio-connector/

**Reply by paulsp94** [reply]
The CSV doesn't export the custom events for me and in the API call I can only get on prop at the time and then I cant match them with each other. I tried calling it like that: 
```
{{BASE_URL}}/api/v1/stats/breakdown?site_id={{SITE_ID}}&period=7d&date={{today}}&interval=date&property=event:props:time
```
and get 
```
{
    "visitors": 1,
    "time": "1096"
}
```
objects. But I want to have an object with all properties from my custom event.
Is there any way to achieve this?

**Reply by RobertJoonas** [reply]
Hi @paulsp94!

It's unfortunately not possible to match different event properties with one another at this stage. So for example, currently you cannot break down by one property and filter on another. This is due to a database limitation.

To support this, we would need to use a [JSON datatype](https://clickhouse.com/docs/en/sql-reference/data-types/json/), which is currently experimental in ClickHouse. Once it's out stable, we are able to start tackling this on our side, but I cannot make any promises on our own development capacity either. We are a small team and need to take things one step at a time. Thanks for understanding!

**Reply by wadjeroudi** [reply]
@RobertJoonas a new json type has been released https://github.com/ClickHouse/ClickHouse/pull/66444
https://clickhouse.com/docs/en/sql-reference/data-types/newjson
do you think you will use it ? Thank you.

**Comment by Hricha-Shandily** [comment]
We have now built an official Plausible Analytics Looker Studio Connector. All details available in our setup guide: https://plausible.io/docs/looker-studio
